src/components/chat-message.tsx

```diff
@@ -36,69 +55,47 @@ const InlineCLI = ( { output, status, time }: InlineCLIProps ) => (
 	</div>
 );
 
-const ActionButton = ( {
-	primaryLabel,
-	secondaryLabel,
-	icon,
-	onClick,
-	timeout,
-	disabled,
-}: {
-	primaryLabel: string;
-	secondaryLabel: string;
-	icon: JSX.Element;
-	onClick: () => void;
-	timeout?: number;
-	disabled?: boolean;
-} ) => {
-	const [ buttonLabel, setButtonLabel ] = useState( primaryLabel );
-
-	const handleClick = () => {
-		onClick();
-		setButtonLabel( secondaryLabel );
-		if ( timeout ) {
-			setTimeout( () => {
-				setButtonLabel( primaryLabel );
-			}, timeout );
-		}
-	};
-
-	return (
-		<Button
-			onClick={ handleClick }
-			variant="outlined"
-			className="h-auto mr-2 !px-2.5 py-0.5 font-sans select-none"
-			disabled={ disabled }
-		>
-			{ icon }
-			<span className="ml-1">{ buttonLabel }</span>
-		</Button>
-	);
-};
+export const ChatMessage = ( {
+	children,
+	id,
+	messageId,
+	isUser,
+	className,
+	projectPath,
+	blocks,
+	updateMessage,
+}: ChatMessageProps ) => {
+	const CodeBlock = ( props: JSX.IntrinsicElements[ 'code' ] & ExtraProps ) => {
+		const content = String( props.children ).trim();
+		const containsWPCommand = /\bwp\s/.test( content );
+		const wpCommandCount = ( content.match( /\bwp\s/g ) || [] ).length;
+		const containsSingleWPCommand = wpCommandCount === 1;
+		const containsAngleBrackets = /<.*>/.test( content );
```

Comment:
We can add more conditions to make this code block only available in a few commands.

Reply:
I noticed that the AI also uses braces `{}` for argument placeholders. I propose including this symbol in the regular expression. WDYT?

<img width="542" alt="Screenshot 2024-06-26 at 11 50 32" src="https://github.com/Automattic/studio/assets/14905380/1dc3cfab-d921-4d1b-9438-29755650fd40">

Reply:
I created [a PR](https://github.com/Automattic/studio/pull/309) with improvements in this area.

Reply:
Noting that I have also seen the Assistant exclude wrapping characters altogether for placeholder content, which complicates things a good bit. I suppose the only solution for this case would be adjusting the Assistant prompt in hopes that it avoid this approach or searching for `example` terms (which undoubtedly is brittle). 

<img width="543" alt="Screenshot 2024-06-26 at 09 10 58" src="https://github.com/Automattic/studio/assets/438664/41e4ab8a-196d-4b47-8b1c-13db9b621b3e">

Reply:
Oh, I haven't encountered that but yes, that definitely complicates the detection of placeholders. As you pointed out, seems the best approach would be to tweak the prompt to ensure it uses the same placeholder to avoid cases like this one.